PWGHF/Tasks/taskD0parametrizedPID.cxx

```diff
@@ -113,13 +118,44 @@ struct TaskD0parametrizedPIDMC {
       //if (ncontributor<=centralitySelectionMin && ncontributor>centralitySelectionMax) {
       //  continue;
       //}
+      Double_t maxFiducialY = 0.8;
+      Double_t minFiducialY = -0.8;
       if (std::abs(particle.flagMCMatchGen()) == 1 << DecayType::D0ToPiK) {
         if (std::abs(RecoDecay::Y(array{particle.px(), particle.py(), particle.pz()}, RecoDecay::getMassPDG(particle.pdgCode()))) > 4.0) {
           continue;
         }
         auto ptGen = particle.pt();
         auto yGen = RecoDecay::Y(array{particle.px(), particle.py(), particle.pz()}, RecoDecay::getMassPDG(particle.pdgCode()));
         registry.fill(HIST("hMassGen"), ptGen, std::abs(yGen));
+        if (ptGen < 5.0) {
+          maxFiducialY = -0.2 / 15 * ptGen * ptGen + 1.9 / 15 * ptGen + 0.5;
+          minFiducialY = 0.2 / 15 * ptGen * ptGen - 1.9 / 15 * ptGen - 0.5;
+        }
+        float etaProngs[2];
+        for (int iD = particle.daughter0Id(), counter = 0; iD <= particle.daughter1Id(); ++iD, counter++) {
+          auto daught = particlesMC.iteratorAt(iD);
+          etaProngs[counter] = std::abs(daught.eta());
+        }
```

Comment:
Why do you need an array and a for loop for getting 2 eta values?

Reply:
I have fixed it and now avoid for loop